pkg/operatorsource/configuring.go

```diff
@@ -95,20 +131,86 @@ func (r *configuringReconciler) Reconcile(ctx context.Context, in *marketplace.O
 	}
 
 	// If we are here, the given CatalogSourceConfig object already exists.
-	cscNamespacedName := types.NamespacedName{Name: in.Name, Namespace: in.Namespace}
-	cscExisting := marketplace.CatalogSourceConfig{}
-	err = r.client.Get(ctx, cscNamespacedName, &cscExisting)
+	err = r.updateExistingCatalogSourceConfig(ctx, in, packages)
 	if err != nil {
-		r.logger.Errorf("Unexpected error while getting CatalogSourceConfig: %s", err.Error())
+		r.logger.Errorf("Unexpected error while updating CatalogSourceConfig: %s", err.Error())
 		nextPhase = phase.GetNextWithMessage(phase.Configuring, err.Error())
-
 		return
 	}
 
+	r.logger.Info("CatalogSourceConfig object has been updated successfully")
+
+	nextPhase = phase.GetNext(phase.Succeeded)
+	return
+}
+
+// getManifestMetadata gets the package metadata from the OperatorSource endpoint
+// it returns the list of packages to be written to the OperatorSource status and an error
```

Comment:
`And an error` sounds confusing. Is it returning the list of the packages AND and error every time? Or is it returning the list of packages, or an error if something goes wrong, which is the pattern used throughout the code base?

Reply:
The first one. The error will be `nil` if there was no error.

Not sure about your statement that this is a pattern used throughout the code base. How would you return an error `or` a result (both being of different types) in go?

Reply:
@aravindhp or @kevinrizza is there a pattern regarding the returned value/error that we are following for these types of helper methods?

Reply:
I think @anik120 was asking clarification regarding when an error is returned. 
```
// getManifestMetadata gets the package metadata from the OperatorSource endpoint.
// It returns the list of packages to be written to the OperatorSource status. error is set
// when there is an issue downloading the metadata. In that case the list of packages
// will be empty.
```
Does this make sense? Or are you questioning the pattern itself?

Reply:
Yup I was asking for a clearer clarification in the comment about when an error is returned. @aravindhp's snippet provides that clarification.

Reply:
Gotcha thanks for helping clarify!